geopandas/geodataframe.py

```diff
@@ -671,6 +671,24 @@ def explode(self):
         geo_df = df.set_geometry(self._geometry_column_name)
         return geo_df
 
+    #overrides the pandas.core.generic.NDFrame method
+    def astype(self, dtype, copy=True, errors='raise', **kwargs):
+        """
+        Temporary overwrite of pandas native astype method. Returns a
+        GeoDataFrame when geometries are valid and a pandas DataFrame when
+        the geometries are invalid.
+
+        Returns
+        -------
+        GeoDataFrame or DataFrame
+        """
+        df = super(DataFrame, self).astype(dtype, copy=copy, errors=errors, **kwargs)
+
+        try:
+            df = geopandas.GeoDataFrame(df, geometry=self._geometry_column_name)
+            return df
+        except TypeError:
+            return df
```

Comment:
I think I found the problem: you need to convert here to a normal dataframe, as pandas' `astype` will sometimes return a plain dataframe, but sometimes also a GeoDataFrame. So you can do a `return pd.DataFrame(df)` on this line.

Reply:
I implemented your suggested solution and a regular pandas DataFrame is returned when the `astype` method is called on the geometry column now.